### ISSUE TITLE
Author will not continue maintenance until next year, refuses to add contributor

### ISSUE THREAD

**Issue by MattMcFarland** [esdoc/esdoc#480]
See his tweet: https://twitter.com/h13i32maru/status/928374936061673472

![image](https://user-images.githubusercontent.com/623545/32630099-7094e4bc-c569-11e7-9a73-2de1d539b411.png)

My question is, do we have any other maintainers that can review PR and triage issues?  This is a great project, but my fear is that its really not sustainable if it only has one maintainer.  I'm really open to maintain the project myself to be honest, but I figure there are probably more qualified people that have contributed in the past.  

What does the community think about this?  Do we want to have more maintainers?  Is it ready for such a thing or am I being neurotic about this.

**Comment by MattMcFarland** [comment]
Some things that really need to be done:

1. There are 4 PR, some of which open for a long time that should be reviewed
2. Issues need triaged, there are a number of issues that should be assigned, closed, or followed up on.
3. esdoc has had **3,170** downloads in the last day, **18,496** downloads in the last week, and 
**81,585** downloads in the last month. With stats like this, shouldn't we have more than one maintainer?

**Comment by iursevla** [comment]
My opinion is that only one mantainer is not enough for a project like. The main maintainer probably has a lot more to do than look for PRs and add new features, bugs etc. 
This are my two cents. I can't help much in this project because  i lack the capabilites for such a project.

But on the other hand i'm a daily user of ESDoc and personally it is my favorite hands down. Even trying to use it for typescript right now.
So i hope this project is maintained well into the future because i'm using it in multiple projects and like the  results.

**Comment by MattMcFarland** [comment]
I love esdoc too @iursevla 

Here are some options I'm thinking of (let me know what you think everyone)

Current options I'm open to:


1. I become a maintainer - I would do my best to honor this community and would triage all of the issues and review PR
1. Someone else is added as maintainer, maybe closer to the author or proven themselves - they would do their best to honor this community, triage all of the issues, review and merge PR.
1. I fork this and republish it and plugins to npm, fixing the issues here and doing the PR. 

Thoughts?

**Comment by iursevla** [comment]
I'd say 1 and 2 are the best options because people already know the ESDoc, if forked it would take some time to get traction. 
Anyway i looked over your account and looks like you have some experience with open source projects, so i think you would do great helping ESDoc.

**Comment by MattMcFarland** [comment]
@h13i32maru since youre away until next year at the least, can you please help us come up with a solution to this problem?

**Comment by MattMcFarland** [comment]
I'd be more than happy to maintain the project if @h13i32maru can add me as a maintainer.  I would triage all of the issues, review PRs and help keep this community alive.

I am trying to reach out to him on twitter, and he hasnt replied yet since I asked him.

If you would like me to maintain the project, or anyone else for that matter, please consider reaching out to @h13i32maru on twitter, we have a conversation you could possibly join. 

conversation here: https://twitter.com/docodemore/status/928967030325342208

**Comment by MattMcFarland** [comment]
https://twitter.com/h13i32maru/status/928969172188459008
"I'm sorry, I are not thinking about adding a maintainer for now. Because I want to advance at my own pace."

This might mean I have to fork and create a new project.  I do not want to do this alone though.

**Comment by raveclassic** [comment]
I gave up _long ago_ and switched entirely to typescript and typedoc. Static typing is _much_ better than JSDoc comments.

**Comment by xeoneux** [comment]
@MattMcFarland why don't you fork it under esdoc2/esdoc2 and continue development? That would be really helpful 😺

**Comment by MattMcFarland** [comment]
@xeoneux I am going to probably do this, but I want to give the author time to reconsider, maybe we can convince him?

**Comment by adilek** [comment]
Let's wait few days if the author will re-think. His response, on Twitter, sounds unserious.
If he still wishes to maintain alone, it would be better to fork.

**Comment by raveclassic** [comment]
@adilek 
> Let's wait few days if the author will re-think.

He won't.

**Comment by raveclassic** [comment]
@adilek There've been various attempts to bring in new feature to jsdoc but all of them were rejected by the author due to lack of interest/time. Check #310. All this led to a fork https://github.com/typhonjs-node-tjsdoc/tjsdoc/issues/2

**Comment by atsushieno** [comment]
Stop spamming people. I didn't want to be spammed for just one single contribution. You could have reviewed before sending spams.

Anyhow I'm 99% sure the author is positive about forked developments. We do create organizations for several reasons (how one developer manages projects), not only to present that "the project is open and welcomes contributions". There is nothing wrong with creating it. You should feel free to fork it, without spamming a lot again.

**Comment by MattMcFarland** [comment]
I'm sorry, I wanted to get everyone's attention because I think this is
detrimental to the community..  My concern is for the 3k downloads a day
and all of the users who are using a project that is paused.  I believe
they deserve better.

I think that it's a terrible mistake to treat such a big project that it's
just some ones personal code. If that's yhe case, Ryo should at least have
the common decency to put in the Readme that this is his own personal
project and he's just sharing it. His "organization" has nothing to do with
the definition of organization but just so he could put multiple projects
together.

**Comment by amercier** [comment]
Hi, I'm not a big contributor (only tiny commit), but anyway, here's my two cents.

I think that whoever wants to to contribute to any open source project should favour opening/fixing issues on the main project (as opposed as his own fork) as possible. This is what you did @MattMcFarland by proposing your help, and that was the right thing to do.

However, the owner of the project clearly stated that he wants to remain the only maintainer of the project, and, where this is not ideal in the sake of de-duplicating effort, it's his right. Having created a separate organisation or not, having said it explicitly in the first place or not, do not change that IMHO.

This leaves contributors and users of the project with two choices: (red pill) create/maintain a fork, or (blue pill) leave as it is and wait for the owner to advance at his own pace. Personally, I don't have enough time for the red pill, so I'll go with the blue one.

**Comment by pzuraq** [comment]
@atsushieno you can unsubscribe if you'd like, it's not unreasonable to try to get people who might care's attention (I'm happy he tagged me, for instance).

@MattMcFarland I agree with all your concerns, this project is growing in usage and quickly becoming a pretty integral part of the infrastructure of the web. JSDoc, YUIDoc, and other solutions have fallen behind significantly and I honestly don't think they're going to catch up. @raveclassic made a great point about TypeDoc, and while I do think that my company will be switching in the future, it's likely going to take time, and I think that there will always be plain ES users out there in the world who could use ESDoc.

That said, I understand where @h13i32maru is coming from. This is their personal project they've worked a long time on, and they probably have a vision for where it should go, what it should do, etc. They probably see some PRs and don't merge because it is counter to that goal, so in the end I think the main thing here that is lacking is communication.

I think we can work together to get past this. I'm willing to put in the time when I can to gain trust, and I'm sure others are too. It could start by:

* Communicating a long term plan for ESDoc - what are the next steps @h13i32maru is planning? What would they like us to build? If we know what their vision is, we can submit PRs that are more inline with that vision directly and thus more likely to be merged, and we can build trust over time
* Working with repeat contributors to build a core - A lot of this is about trust, as I said before I'm sure @h13i32maru doesn't want to add contributors who are just going to merge things left and right without consulting them. I personally never merge a PR on a project I have commit on unless I'm sure the original author will be ok with it, and I'm sure there are plenty of maintainers out there who would be respectful of @h13i32maru's intent. If that intent is communicated well, they can continue to work when @h13i32maru is busy as well

If we can't build toward something here, a fork is pretty much inevitable. I think that would be horrible - it would divide the nascent community, and it would divide developer talent and time -
 but it may be necessary for everyone's best interest. ESDoc has already done a ton of work getting people's attention, and a ton of people have placed their trust in @h13i32maru, so whether they consider it a community project or not, at this point it is one.

**Comment by raveclassic** [comment]
@pzuraq Great words! I didn’t try to seem impolite and really hope all the work that’s done so far won’t go down the drain. I’m just trying to tell that the best decision for _my team_ was to switch to TS. 
Still I’m looking forward to further development of esdoc no matter which way the community around it is going to pick.

**Comment by MattMcFarland** [comment]
Since the @h13i32maru is not interested in adding any contributor to the project, I have forked esdoc to esdoc2 - https://github.com/esdoc2/esdoc2

I will now migrate all issues over and accepting PRs.
I want esdoc2 to be a community effort, not my own personal effort like @h13i32maru - if youre interested in joining the esdoc2 org let me know.  

As of now my roadmap for esdoc2 is to bring awareness to users of esdoc that there is one that will be community focused.   Ownership of esdoc2 **will not be centralized** - I have finished porting all code over to esdoc2 as published on npm.

I will be porting over all outstanding issues, feel free to send a PR on the esdoc2 branch.

**Comment by raveclassic** [comment]
@MattMcFarland Please contact @typhonrt. There’s a plenty of work done around migration to latest dev stack. I hope it could be merged to a truly community-driven library.

**Comment by atsushieno** [comment]
Never mention me again, I turned off once but since someone mentioned me and then github turned on notification again.

**Comment by MattMcFarland** [comment]
Author has returned, I'm thinking the language barrier must be an issue.
